File: 1844_replace_all_digits_with_characters/Solution.kt
// LeetCode 1844 - Replace All Digits with Characters
// https://leetcode.com/problems/replace-all-digits-with-characters/

class Solution {
    fun replaceDigits(s: String): String {
        val chars = s.toCharArray()
        var i = 1
        while (i < chars.size) {
            chars[i] = (chars[i - 1].code + (chars[i] - '0')).toChar()
            i += 2
        }
        return String(chars)
    }
}

File: 1844_replace_all_digits_with_characters/Solution.cs
// LeetCode 1844 - Replace All Digits with Characters
// https://leetcode.com/problems/replace-all-digits-with-characters/

public class Solution {
    public string ReplaceDigits(string s) {
        char[] chars = s.ToCharArray();
        for (int i = 1; i < chars.Length; i += 2) {
            chars[i] = (char)(chars[i - 1] + (chars[i] - '0'));
        }
        return new string(chars);
    }
}
